tools/RunGen.cpp

```diff
@@ -498,17 +498,17 @@ bool adapt_input_buffer_layout(const Shape &constrained_shape, Buffer<> *buf) {
         fail() << "Dimension mismatch";
     }
     for (size_t i = 0; i < constrained_shape.size(); ++i) {
-        // min of nonzero means "min"
+        // min of nonzero means "largest value for min"
```

Comment:
Not necessarily. That may be the typical use, but it could also be that the min is required to be odd, or same as the min in some other dimension. The same comment applies to non-zero extents.